pkg/sinks/bigquery.go

```diff
@@ -0,0 +1,225 @@
+package sinks
+
+import (
+	"bufio"
+	"cloud.google.com/go/bigquery"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/opsgenie/kubernetes-event-exporter/pkg/batch"
+	"github.com/opsgenie/kubernetes-event-exporter/pkg/kube"
+	"github.com/rs/zerolog/log"
+	"google.golang.org/api/option"
+	"os"
+	"time"
+	"unicode"
+)
+
+// Returns a map filtering out keys that have nil value assigned.
+func dropNils(x map[string]interface{}) map[string]interface{} {
+	y := make(map[string]interface{})
+	for key, value := range x {
+		if value != nil {
+			if mapValue, ok := value.(map[string]interface{}); ok {
+				y[key] = dropNils(mapValue)
+			} else {
+				y[key] = value
+			}
+		}
+	}
+	return y
+}
+
+// Returns a string representing a fixed key. BigQuery expects keys to be valid identifiers, so if they aren't we modify them.
+func fixKey(key string) string {
+	var fixedKey string
+	if !unicode.IsLetter(rune(key[0])) {
+		fixedKey = "_"
+	}
+	for _, ch := range key {
+		if unicode.IsLetter(ch) || unicode.IsDigit(ch) {
+			fixedKey = fixedKey + string(ch)
+		} else {
+			fixedKey = fixedKey + "_"
+		}
+	}
+	return fixedKey
+}
+
+// Returns a map copy with fixed keys.
+func fixKeys(x map[string]interface{}) map[string]interface{} {
+	y := make(map[string]interface{})
+	for key, value := range x {
+		if mapValue, ok := value.(map[string]interface{}); ok {
+			y[fixKey(key)] = fixKeys(mapValue)
+		} else {
+			y[fixKey(key)] = value
+		}
+	}
+	return y
+}
```

Comment:
As the `sinks` package is shared with lots of others, likely going to run into some namespacing issues with these function names. May prefix them with `bigQuerySinkBlah` or better yet attach them to the `BigQuery` struct so they are just on that object itself.